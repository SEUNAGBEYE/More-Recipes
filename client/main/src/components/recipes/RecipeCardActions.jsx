import React from 'react';
import { Link } from 'react-router-dom';
import classnames from 'classnames';

const RecipeCardAction = (props) => (
  <div>
    { props.recipe.userId === props.user.userId ?
      <div className="d-flex justify-content-between recipe-icons" style={props.style}>
        <i className="fa fa-eye icons">{props.recipe.views.length}</i>
        <Link to="/" className="fa fa-pencil icons" data-toggle="modal" data-target={`#editModal${props.recipe.id}`}/>
        <Link to="/" className="fa fa-trash icons" data-toggle="modal" data-target={`#deleteModal${props.recipe.id}`} />
      </div> :
      <div className="d-flex justify-content-between recipe-icons" style={props.style}>
        <i className="fa fa-eye icons">{props.recipe.views.length}</i>
        <i className={classnames("fa icons", {
          'fa-thumbs-o-down text-black': !props.isDownVoted,
          'fa-thumbs-down text-warning': props.isDownVoted
        })}
        onClick={props.toggleThumbsDownRecipe} id={props.recipe.id} >{props.recipe.downvotes ? props.recipe.downvotes.length : 0}</i>

        <i className={classnames("fa icons", {
          'fa-thumbs-o-up text-black': !props.isUpVoted,
          'fa-thumbs-up text-warning': props.isUpVoted
        })}
        onClick={props.toggleThumbsUpRecipe} id={props.recipe.id} >{props.recipe.upvotes ? props.recipe.upvotes.length : 0}</i>

        <i
          className={classnames(
            "fa icons",
            {
              "fa-heart-o text-black": !props.isFavorited,
              "fa-heart text-warning": props.isFavorited
            }
          )}
          onClick={props.toggleFavouriteRecipe}
          id={props.recipe.id} />

      </div>
    }
  </div>
);

export default RecipeCardAction;
