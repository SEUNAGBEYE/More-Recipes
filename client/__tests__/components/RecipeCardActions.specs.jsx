

import React from 'react';
import { shallow } from 'enzyme';
import RecipeCardActions from '../../main/src/components/recipes/RecipeCardActions';

const recipe = {
  userId: 1,
  name: 'Amala',
  image: 'This is the image',
  upvotes: [1,2,4],
  downvotes: [1,2,4],
  description: 'This is the decription',
  views: [1,2]
}

const props = {
  recipe,
  user: {
    firstName: 'Seun',
    lastName: 'Agbeye'
  },
  isDownVoted: true,
  isUpVoted: true,
  isFavorited: true,
  toggleFavouriteRecipe: jest.fn(),
  toggleThumbsDownRecipe: jest.fn(),
  toggleThumbsUpRecipe: jest.fn(),
};

const event = {
  preventDefault: jest.fn(),
  target: {
    id: 1
  }
}

describe('# RecipeCardActions', () => {
  const wrapper = shallow(
      <RecipeCardActions {...props} />
  );

  it('should render successfully', (done) => {
    expect(wrapper).toBeDefined();
    expect(wrapper.getElement().type).toBe('div');
    expect(wrapper).toMatchSnapshot();
    done();
  });
});
