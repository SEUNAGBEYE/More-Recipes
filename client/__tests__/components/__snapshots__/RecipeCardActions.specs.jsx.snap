// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`# RecipeCardActions should render successfully 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RecipeCardActions
    isDownVoted={true}
    isFavorited={true}
    isUpVoted={true}
    recipe={
      Object {
        "description": "This is the decription",
        "downvotes": Array [
          1,
          2,
          4,
        ],
        "image": "This is the image",
        "name": "Amala",
        "upvotes": Array [
          1,
          2,
          4,
        ],
        "userId": 1,
        "views": Array [
          1,
          2,
        ],
      }
    }
    toggleFavouriteRecipe={[MockFunction]}
    toggleThumbsDownRecipe={[MockFunction]}
    toggleThumbsUpRecipe={[MockFunction]}
    user={
      Object {
        "firstName": "Seun",
        "lastName": "Agbeye",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="d-flex justify-content-between recipe-icons"
        style={undefined}
      >
        <i
          className="fa fa-eye icons"
        >
          2
        </i>
        <i
          className="fa icons fa-thumbs-down text-warning"
          id={undefined}
          onClick={[MockFunction]}
        >
          3
        </i>
        <i
          className="fa icons fa-thumbs-up text-warning"
          id={undefined}
          onClick={[MockFunction]}
        >
          3
        </i>
        <i
          className="fa icons fa-heart text-warning"
          id={undefined}
          onClick={[MockFunction]}
        />
      </div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <i
            className="fa fa-eye icons"
          >
            2
          </i>,
          <i
            className="fa icons fa-thumbs-down text-warning"
            id={undefined}
            onClick={[MockFunction]}
          >
            3
          </i>,
          <i
            className="fa icons fa-thumbs-up text-warning"
            id={undefined}
            onClick={[MockFunction]}
          >
            3
          </i>,
          <i
            className="fa icons fa-heart text-warning"
            id={undefined}
            onClick={[MockFunction]}
          />,
        ],
        "className": "d-flex justify-content-between recipe-icons",
        "style": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": 2,
            "className": "fa fa-eye icons",
          },
          "ref": null,
          "rendered": 2,
          "type": "i",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": 3,
            "className": "fa icons fa-thumbs-down text-warning",
            "id": undefined,
            "onClick": [MockFunction],
          },
          "ref": null,
          "rendered": 3,
          "type": "i",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": 3,
            "className": "fa icons fa-thumbs-up text-warning",
            "id": undefined,
            "onClick": [MockFunction],
          },
          "ref": null,
          "rendered": 3,
          "type": "i",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "fa icons fa-heart text-warning",
            "id": undefined,
            "onClick": [MockFunction],
          },
          "ref": null,
          "rendered": null,
          "type": "i",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="d-flex justify-content-between recipe-icons"
          style={undefined}
        >
          <i
            className="fa fa-eye icons"
          >
            2
          </i>
          <i
            className="fa icons fa-thumbs-down text-warning"
            id={undefined}
            onClick={[MockFunction]}
          >
            3
          </i>
          <i
            className="fa icons fa-thumbs-up text-warning"
            id={undefined}
            onClick={[MockFunction]}
          >
            3
          </i>
          <i
            className="fa icons fa-heart text-warning"
            id={undefined}
            onClick={[MockFunction]}
          />
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <i
              className="fa fa-eye icons"
            >
              2
            </i>,
            <i
              className="fa icons fa-thumbs-down text-warning"
              id={undefined}
              onClick={[MockFunction]}
            >
              3
            </i>,
            <i
              className="fa icons fa-thumbs-up text-warning"
              id={undefined}
              onClick={[MockFunction]}
            >
              3
            </i>,
            <i
              className="fa icons fa-heart text-warning"
              id={undefined}
              onClick={[MockFunction]}
            />,
          ],
          "className": "d-flex justify-content-between recipe-icons",
          "style": undefined,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": 2,
              "className": "fa fa-eye icons",
            },
            "ref": null,
            "rendered": 2,
            "type": "i",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": 3,
              "className": "fa icons fa-thumbs-down text-warning",
              "id": undefined,
              "onClick": [MockFunction],
            },
            "ref": null,
            "rendered": 3,
            "type": "i",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": 3,
              "className": "fa icons fa-thumbs-up text-warning",
              "id": undefined,
              "onClick": [MockFunction],
            },
            "ref": null,
            "rendered": 3,
            "type": "i",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "fa icons fa-heart text-warning",
              "id": undefined,
              "onClick": [MockFunction],
            },
            "ref": null,
            "rendered": null,
            "type": "i",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
